File: Backend/Scheduler/Model/OrderReport/OrderReport.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Scheduler.Model.OrderReport
{
    public class OrderReport
    {
        public int Id { get; set; }

        public int OrderId { get; set; }
        public virtual Order Order { get; set; }

        public DateTime CreationTime { get; set; }

        public virtual List<OrderBlock> OrderBlocks { get; set; }
    }
}

File: Frontend/HttpClient/src/app/models/Reporting/OrderReport.ts
import { OrderBlock } from './OrderBlock';
import { Order } from '../order';

export class OrderReport {
    public id: number;

    public creationTime: Date;

    public orderBlocks: OrderBlock[];

    public order: Order;

    static fromJSON(obj: any) {
        if (!obj) {
            return null;
        }

        return Object.assign(
            new OrderReport(),
            obj,
            {
                orderBlocks: obj.orderBlocks ? obj.orderBlocks.map(g => OrderBlock.fromJSON(g)) : []
            }
        );
    }
}
